framework/test/unit/modules/interoperability/mainchain/module.spec.ts

```diff
@@ -779,6 +779,110 @@ must NOT have more than ${MAX_NUM_VALIDATORS} items`,
 				await expect(interopMod.initGenesisState(context)).resolves.not.toThrow();
 			});
 
+			it('should throw if chainInfo.chainData.status===TERMINATED exists but no terminateStateAccount', async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...genesisInteroperability,
+						chainInfos: [
+							{
+								...chainInfo,
+								chainData: {
+									...chainData,
+									status: ChainStatus.TERMINATED,
+									lastCertificate: {
+										...lastCertificate,
+										validatorsHash: computeValidatorsHash(activeValidators, certificateThreshold),
+									},
+								},
+								chainValidators: {
+									activeValidators,
+									certificateThreshold,
+								},
+							},
+						],
+						// No terminatedStateAccount
+						terminatedStateAccounts: [],
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`For each terminatedStateAccount there should be a corresponding chainInfo at TERMINATED state`,
+				);
+			});
+
+			it('should throw if there is an entry in terminateStateAccounts for a chainID that is ACTIVE in chainInfos', async () => {
+				const context = createInitGenesisStateContext(
+					{
+						...genesisInteroperability,
+						chainInfos: [
+							{
+								...chainInfo,
+								chainData: {
+									...chainData,
+									status: ChainStatus.ACTIVE,
+									lastCertificate: {
+										...lastCertificate,
+										validatorsHash: computeValidatorsHash(activeValidators, certificateThreshold),
+									},
+								},
+								chainValidators: {
+									activeValidators,
+									certificateThreshold,
+								},
+							},
+						],
+						terminatedStateAccounts: [
+							{
+								chainID: chainInfo.chainID,
+								terminatedStateAccount,
+							},
+						],
+					},
+					params,
+				);
+
+				await expect(interopMod.initGenesisState(context)).rejects.toThrow(
+					`For each terminatedStateAccount there should be a corresponding chainInfo at TERMINATED state`,
+				);
+			});
+
+			it('should throw if there is no entry for terminateStateAccount chainID in chainInfos', async () => {
```

Comment:
I think this is already checked (test of line 922), so it could be removed. But better do it twice than zero.. ;-)